### PR TITLE
BUG: guard against replacing constants without `'_'` spec

### LINKED ISSUE #8493: f2py: parameter type inference fails because incomplete handling of kind specs
The following valid code snippet originally from pymc/gibbsit.f  (cf  https://github.com/pymc-devs/pymc/blob/master/pymc/gibbsit.f )  exhibits the problem: 
```
...
      program   gibbsit

      integer   maxiterates
      parameter (maxiterates=50000)

      integer   maxseries
      parameter (maxseries=20)

      integer   maxqcnt
      parameter (maxqcnt=20)

      integer   wasize
      parameter (wasize=maxiterates*2)
```
showing up for f2py version 2 in numpy 1.12.0 as: 
```
   Line #196 in pymc/gibbsit.f:"      parameter (wasize=maxiterates*2)"
        get_parameters: got "unexpected EOF while parsing (<string>, line 0)" on ''
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-en5WK5/pymc/setup.py", line 124, in <module>
        **(config_dict))
      File "/usr/local/lib/python2.7/dist-packages/numpy/distutils/core.py", line 169, in setup
        return old_setup(**new_attr)
      File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/local/lib/python2.7/dist-packages/numpy/distutils/command/egg_info.py", line 18, in run
        self.run_command("build_src")
      File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/local/lib/python2.7/dist-packages/numpy/distutils/command/build_src.py", line 148, in run
        self.build_sources()
      File "/usr/local/lib/python2.7/dist-packages/numpy/distutils/command/build_src.py", line 165, in build_sources
        self.build_extension_sources(ext)
      File "/usr/local/lib/python2.7/dist-packages/numpy/distutils/command/build_src.py", line 327, in build_extension_sources
        sources = self.f2py_sources(sources, ext)
      File "/usr/local/lib/python2.7/dist-packages/numpy/distutils/command/build_src.py", line 564, in f2py_sources
        ['-m', ext_name]+f_sources)
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/f2py2e.py", line 408, in run_main
        postlist = callcrackfortran(files, options)
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/f2py2e.py", line 329, in callcrackfortran
        postlist = crackfortran.crackfortran(files)
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/crackfortran.py", line 3245, in crackfortran
        readfortrancode(files, crackline)
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/crackfortran.py", line 511, in readfortrancode
        dowithline(finalline)
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/crackfortran.py", line 782, in crackline
        analyzeline(m, pat[1], line)
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/crackfortran.py", line 1188, in analyzeline
        t = determineexprtype(initexpr, params)
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/crackfortran.py", line 2971, in determineexprtype
        return _ensure_exprdict(vars[e])
      File "/usr/local/lib/python2.7/dist-packages/numpy/f2py/crackfortran.py", line 2947, in _ensure_exprdict
        raise AssertionError(repr(r))
    AssertionError: ''
```

a pull request with a proposal to fix it will follow.

### PR DESCRIPTION
**Pull request by utke1**
In the reported problem snippet the attempt to infer the type of
    parameter (wasize=maxiterates*2)
leads to a lookup of maxiterates in the logic of get_parameters which finds
and correctly assigns to 'v' the value string '50000'
However, then in the logic proceds to attempt to split off any underscore-appended precision
spec from v_ which yields ['50000']  because there is no underscore.
Finally line
v = ''.join(v_[:-1]).lower().replace(v_[-1].lower(), '')
if not quarded by the newly introduced lengthtest sets v to empty because
[:-1]  of a list with one element yields an empty list.
Subsequently the type inference by running eval on an empty string fails with the error
message quoted.
The introduced length check appears to correct this problem.

### CODE REVIEW

**Comment by charris** [comment]
Needs tests, should also test if this fixes #8479. Also read `doc/source/dev/gitwash/development_workflow.rst` on formatting commit messages. Note that they need hard line breaks and the lines should be <= 72 characters long.

**Comment by charris** [comment]
And the `Fixes #8493` should be part of the commit message rather than the summary line.

**Comment by charris** [comment]
You can see some tests at #8299. @zerothi Ping.

**Comment by utke1** [comment]
To @charris  - when you commented on the formatting of the commit message - do you require me to re-commit the change (i.e. kill this pull request and make a new one) or will you accept the commit message as is and I will just add some test cases into this pull request?

**Comment by charris** [comment]
You can edit commit messages, squash commits, and do other good things using `git commit -i HEAD~n`, where n is the number of recent commits to edit, followed by a force push `git push -f origin HEAD`.

**Comment by pv** [comment]
@charris probably meant `git rebase -i HEAD~n` not `git commit ...`.

**Comment by madphysicist** [comment]
Also, keep in mind that the PR is tied to the branch name, so any sequence of commits you assign the name to will be the subject of this PR. Modify your git history at will with rebasing, cherry-picking or outright branch renaming with that in mind.

**Comment by charris** [comment]
Ah, I see that you are also working in the master branch, big no no. You can fix that by checking out a work branch `git co -b <new branch>`, then resetting master to before your work `git reset --hard <hash of last valid commit>. You will need to close this PR and make a new one.

**Comment by zerothi** [comment]
I have went through the code and it looks good to me, thanks for fixing my mistake. I have added a test which you may grab here:
<script src="https://gist.github.com/zerothi/d7f7beb5064be69281a3b0ede97a769d.js"></script>

**Comment by utke1** [comment]
Changed commit message and pull request title to match top line commit message

**Comment by utke1** [comment]
To @charris  - with the 2 tests included above - this is all I was going to do to amend this pull request - please let me know if there is anything else you need me to do before you can reconsider merging.

File: numpy/f2py/crackfortran.py [hunk 1]
```diff
@@ -2451,7 +2451,8 @@ def get_parameters(vars, global_params={}):
                 if not selected_kind_re.match(v):
                     v_ = v.split('_')
                     # In case there are additive parameters
-                    v = ''.join(v_[:-1]).lower().replace(v_[-1].lower(), '')
+                    if len(v_)>1:
```
Comment (charris) [inline numpy/f2py/crackfortran.py]: Spaces around `>`.

**Comment by charris** [comment]
Just a small style nitpick. @2sn Does this fix your problem?

File: numpy/f2py/crackfortran.py (hunk 1 above)
Comment (utke1) [inline numpy/f2py/crackfortran.py]: done

**Comment by charris** [comment]
Doesn't seem to be anymore feedback on this, so in it goes. Thanks @utke1 .

**Comment by charris** [comment]
Might want to clean up your master branch now as it will be out of sync with numpy master. Or you could just refork and start over.

**Comment by 2sn** [comment]
Yes, this seems to fix the compile issue. Thanks!

**Pull request merged**